src/Inventory.zig

```diff
@@ -1864,23 +1864,26 @@ fn update(self: Inventory) void {
 	if(self.type == .workbench) {
 		self._items[self._items.len - 1].deinit();
 		self._items[self._items.len - 1].clear();
-		var availableItems: [25]?*const BaseItem = undefined;
+		var availableItems: [25]?BaseItemIndex = undefined;
 		var hasAllMandatory: bool = true;
+		const slotInfos = self.type.workbench.slotInfos;
 
 		for(0..25) |i| {
 			if(self._items[i].item != null and self._items[i].item.? == .baseItem) {
 				availableItems[i] = self._items[i].item.?.baseItem;
 			} else {
-				if(!self.type.workbench.slotInfos[i].optional and !self.type.workbench.slotInfos[i].disabled)
+				if(!slotInfos[i].optional and !slotInfos[i].disabled) {
 					hasAllMandatory = false;
+					break;
```

Comment:
Maybe it would be best to just ditch the `hasAllMandatory`, and just return directly.
I don't like your current approach since it leads to uninitialized data without a guarantee that it isn't used again below.